src/main/java/rs/hybridit/controller/AuthenticationController.java

```diff
@@ -0,0 +1,66 @@
+package rs.hybridit.controller;
+
+import java.io.IOException;
+import javax.servlet.http.HttpServletResponse;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.core.AuthenticationException;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import rs.hybridit.WebSecurityConfig;
+import rs.hybridit.auth.JwtAuthenticationRequest;
+import rs.hybridit.dto.MessageDto;
+import rs.hybridit.dto.UserDto;
+import rs.hybridit.model.UserTokenState;
+import rs.hybridit.security.TokenHelper;
+import rs.hybridit.service.UserService;
+import rs.hybridit.serviceImpl.UserServiceImpl;
+
+@RestController
+@RequestMapping(value = "/auth")
+@AllArgsConstructor
+public class AuthenticationController {
+
+	private TokenHelper tokenUtils;
+
+	private WebSecurityConfig webSecurityConfig;
+
+	private UserService userService;
+
+	@PostMapping(value = "/registerAdmin")
+	@PreAuthorize("hasRole('ADMIN')")
+	public ResponseEntity<?> registerAdmin(@RequestBody UserDto user) {
+		if (this.userService.registerAdmin(user) != null) {
```

Comment:
Register admin returns boolean value which is going to be either true or false, not null. You could throw exception directly in the registerAdmin method and have void method registerAdmin. Same goes for registerLibrarian

Reply:
Then you can wrap this with try / catch and return new ResponseEntity<>(MessageDto(e.getMessage()), "Error"), HttpStatus.CONFLICT);

This will make it easier once we introduce Controller Advices

Reply:
Modified it.